core/src/main/java/org/apache/accumulo/core/fate/zookeeper/ZooCache.java

```diff
@@ -525,6 +526,36 @@ public void clear(String zPath) {
     }
   }
 
+  /**
+   * Removes all paths in the cache match the predicate.
+   */
+  public void clear(Predicate<String> pathPredicate) {
+    Preconditions.checkState(!closed);
+    Predicate<String> pathPredicateToUse;
+    if (log.isTraceEnabled()) {
+      pathPredicateToUse = path -> {
+        boolean testResult = pathPredicate.test(path);
+        if (testResult) {
+          log.trace("removing {} from cache", path);
+        }
+        return testResult;
+      };
```

Comment:
Would this work?

```suggestion
      pathPredicateToUse = pathPredicate.and(path -> {
        log.trace("removing {} from cache", path);
        return true;
      });
```

Reply:
Yes, I think this would work. Based on the javadoc it looks like if `pathPredicate` returns false, then the logging will not occur.

Reply:
Added this change in dc5e2c3